src/views/virtualmachines/list/components/Timestamp/datetime.ts

```diff
@@ -0,0 +1,113 @@
+const maxClockSkewMS = -60000;
+
+const dateTimeFormatterOptions: Intl.DateTimeFormatOptions = {
+  month: 'short',
+  day: 'numeric',
+  hour: 'numeric',
+  minute: 'numeric',
+  year: 'numeric',
+};
+const utcDateTimeFormatterOptions: Intl.DateTimeFormatOptions = {
+  ...dateTimeFormatterOptions,
+  timeZone: 'UTC',
+  timeZoneName: 'short',
+};
+const dateTimeFormatter = new Intl.DateTimeFormat(undefined, dateTimeFormatterOptions);
+
+export const utcDateTimeFormatter = new Intl.DateTimeFormat(undefined, utcDateTimeFormatterOptions);
+
+const relativeTimeFormatter = Intl.RelativeTimeFormat ? new Intl.RelativeTimeFormat() : null;
+
+const getDuration = (ms: number) => {
+  if (!ms || ms < 0) {
+    ms = 0;
+  }
+  let seconds = Math.floor(ms / 1000);
+  let minutes = Math.floor(seconds / 60);
+  seconds = seconds % 60;
+  let hours = Math.floor(minutes / 60);
+  minutes = minutes % 60;
+  const days = Math.floor(hours / 24);
+  hours = hours % 24;
+  return { days, hours, minutes, seconds };
+};
+
+export const isValid = (dateTime: Date): boolean =>
+  dateTime instanceof Date && !isNaN(dateTime.valueOf());
+
+const fromNow = (dateTime: string | Date, now?: Date, options?) => {
+  // Check for null. If dateTime is null, it returns incorrect date Jan 1 1970.
+  if (!dateTime) {
+    return '-';
+  }
+
+  if (!now) {
+    now = new Date();
+  }
+
+  const d = new Date(dateTime);
+  const ms = now.getTime() - d.getTime();
+  const justNow = 'Just now';
+
+  // If the event occurred less than one minute in the future, assume it's clock drift and show "Just now."
+  if (!options?.omitSuffix && ms < 60000 && ms > maxClockSkewMS) {
```

Comment:
Would you mind making a constant for numbers like these? Something like:
`MS_IN_MINUTE` or `NUM_MS_IN_MINUTE`

Reply:
+1 changed thanks :)